### PR TITLE
Update system_design_parameter_schema.json

### PR DESCRIPTION
**Pull request by amyeallen1**
Hi Nick, these are my suggested additions to the system design parameters schema. 

The "optimization approaches" can be expanded in the future to also include specific algorithms. The boolean for "topology optimization" is an indicator for performing topology optimization or not.

### CODE REVIEW

File: geojson_modelica_translator/system_parameters/schema/system_design_parameter_schema.json [hunk 1]
```diff
@@ -76,6 +108,37 @@
       },
       "additionalProperties": false
     },
+    "cost_fun_def": {
```
Comment (nllong) [inline geojson_modelica_translator/system_parameters/schema/system_design_parameter_schema.json]: can you spell out this "cost_function_definition"?

File: geojson_modelica_translator/system_parameters/schema/system_design_parameter_schema.json [hunk 2]
```diff
@@ -76,6 +108,37 @@
...
+          "enum": [
+            "Energy",
+            "Life Cycle Cost",
+            "Carbon Emissions"
+          ]
+        }
+      },
+      "additionalProperties": false
+    },
+    "opt_approach_def": {
```
Comment (nllong) [inline geojson_modelica_translator/system_parameters/schema/system_design_parameter_schema.json]: also spell this out or rename to something shorter (but fully spelled out).

File: geojson_modelica_translator/system_parameters/schema/system_design_parameter_schema.json [hunk 3]
```diff
@@ -76,6 +108,37 @@
...
+      "title": "opt_approach_def",
+      "description": "Optimization approach for the topology optimization problem",
+      "type": "object",
+      "properties": {
+        "opt_approach": {
```
Comment (nllong) [inline geojson_modelica_translator/system_parameters/schema/system_design_parameter_schema.json]: spell out

File: geojson_modelica_translator/system_parameters/schema/system_design_parameter_schema.json [hunk 4]
```diff
@@ -76,6 +108,37 @@
       },
       "additionalProperties": false
     },
+    "cost_fun_def": {
+      "title": "defaults",
+      "description": "Possible cost functions for the topology optimization problem",
+      "type": "object",
+      "properties": {
+        "cost_fun": {
```
Comment (nllong) [inline geojson_modelica_translator/system_parameters/schema/system_design_parameter_schema.json]: spell out

File: geojson_modelica_translator/system_parameters/schema/system_design_parameter_schema.json [hunk 5]
```diff
@@ -50,10 +50,42 @@
...
+            "topology_optimization": {
+              "description": "Perform topology optimization?",
+              "type": "boolean"
+            },
+            "Cost_Function": {
+              "description": "Cost function to use in topology optimization",
+              "type": "string",
+              "$ref": "#/definitions/cost_fun_def"
+            },
+            "Optimization_Approach": {
```
Comment (nllong) [inline geojson_modelica_translator/system_parameters/schema/system_design_parameter_schema.json]: We want these to be snake cased, so remove the capitals on all of the items.

**Comment by amyeallen1** [comment]
Thanks for reviewing this, Nick. I made those revisions and added one other new field, and committed the changes. Do I need to make a new PR?

**Comment by nllong** [comment]
No you don’t need to make a new PR. Simply click the “re-request” review
for me on the PR

Il giorno mer 24 lug 2019 alle 10:15 amyeallen1 <notifications@github.com>
ha scritto:

**Pull request merged**